src/stores/room-list/algorithms/Algorithm.ts

```diff
@@ -281,14 +290,26 @@ export class Algorithm extends EventEmitter {
     protected recalculateFilteredRoomsForTag(tagId: TagID): void {
         console.log(`Recalculating filtered rooms for ${tagId}`);
         delete this.filteredRooms[tagId];
-        const rooms = this.cachedRooms[tagId];
+        const rooms = this.cachedRooms[tagId].map(r => r); // cheap clone
```

Comment:
Is this really better than `.slice()`?

Reply:
It's no worse. This has been debated before, and is at least consistent with the code in the area.

Reply:
Fair enough - agreed it is more idiomatic.